lib/types/results-wrapper.js

```diff
@@ -12,78 +12,75 @@ const Row = require("./row");
 const Tuple = require("./tuple");
 
 /**
- * Checks the type of value wrapper object and gets it from the underlying value
+ * Maps the value returned from the rust driver into expected JS object.
+ *
+ * This function accepts the following kinds of arguments:
+ *   1. `Value` -> In case we can correctly handle the value based on the type;
+ *   2. `[Type, Value]` ->  In case of ambiguity, rust side provides tuple of (CqlType, Value);
+ *   3. `Array` -> Array is provided as value. Because tuple is represented in JS as array,
+ *       we need to distinguish between this and previous type, based on first value type.
```

Comment:
```suggestion
 *   3. `Array` -> Array is provided as value. Because tuple is represented in JS as an array,
 *       we need to distinguish between this and previous type, based on first value type.
```
The explanation for the array is in my opinion complicated and hard to understand.